app.js

```diff
@@ -122,19 +121,12 @@ app.use(function (state, emitter) {
   }
 
   function updateStorageBackend () {
-    var storage = state.storage
-    var currentUrl = storage.getBackend().getRootUrl()
-    var url = getStorageUrl()
-    if (url && url !== currentUrl) {
-      console.info('now using storage url', url)
-      state.storage.setBackend(createStorageBackend(state, url))
-      state.map.draw()
-    }
+    state.storage.updateBackend(state, getStorageUrl())
+    state.map.draw()
   }
 
   function onReady () {
-    var backend = createStorageBackend(state, getStorageUrl())
-    state.storage = createStorage(backend)
```

Comment:
`app.js` knows too much about the backend, it should instead just call update on the storage and it will pick